lib/sqlalchemy/orm/util.py

```diff
@@ -1015,7 +1016,18 @@ def _alias_factory(
         flat: bool = False,
         adapt_on_names: bool = False,
     ) -> Union[AliasedClass[_O], FromClause]:
-        if isinstance(element, FromClause):
+        if isinstance(element, GenerativeSelect):
+            util.warn(
+                f"{type(element).__name__} can't be aliased, "
+                f"coercing to subquery instead."
+            )
+            return coercions.expect(
```

Comment:
how about we instead just raise an informative unsupported error here? This is what other cases like this behave in 2.0 so I don't think it makes much sense to add support for a previously unsupported feature (from by understanding this raised an exception in all cases) and also deprecate it right away